presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -318,4 +324,31 @@ else if (ARRAY.equals(type.getTypeSignature().getBase())) {
         path.remove(name);
         return block;
     }
+
+    /**
+     * Helper used to determine the appropriate group annotation processing markers to use based on the schema of the parquet file being read.
+     *
+     * @param path The path to the parent column. This should be a map column.
+     * @return The value to use for the map group annotation in the column path.
+     */
+    @VisibleForTesting
+    String getMapKeyPath(List<String> path)
+    {
+        Optional<RichColumnDescriptor> descriptor = getDescriptor(fileSchema, requestedSchema, path);
+
+        if (!descriptor.isPresent()) {
+            return LEGACY_MAP_TYPE_NAME;
```

Comment:
Why is legacy the default here? Will this fail if descriptor is empty, but it is in the new/current naming?